### ISSUE TITLE
`no-unused-modules` reports missing exports for `export *`

### ISSUE THREAD

**Issue by robin-hartmann** [import-js/eslint-plugin-import#2414]
The rule `no-unused-modules` incorrectly reports `No exports found` for files which only contain `export * from 'foo'` or `export * as bar from 'buz'` statements, if `missingExports` is set to `true`.

To be precise, all of the following files get reported as `No exports found`:

```ts
export * from 'foo';
```

```ts
export * from 'foo';
export * from 'bar';
```

```ts
export * as foo from 'foo';
```

```ts
export * as foo from 'foo';
export * as bar from 'bar';
```

```ts
export * from 'foo';
export * as bar from 'bar';
```

**Comment by robin-hartmann** [comment]
I've already spent multiple hours trying to fix it myself, but I never managed to get all tests to complete successfully, I only managed to break different ones... but I opened a PR (#2415) to add more tests (which are failing at the moment, of course).

**Comment by robin-hartmann** [comment]
I also just added my current version of the "fixed" tests with only 1 failing test to the PR to gather some input or as a jumping-off point for someone else.
